src/python/dxpy/cli/dataset_utilities.py

```diff
@@ -1043,6 +1048,20 @@ def print_fields(fields):
                 quoting=csv.QUOTE_NONE,
             )
 
+def extract_assay_expression(args):
+    """
+    Retrieve the selected data or generate SQL to retrieve the data from an expression assay in a dataset or cohort based on provided rules.
+    """
+
+    # Validating input combinations
+    input_validator = ExpressionInputsValidator(args)
+    if input_validator.error_handler:
```

Comment:
It would be a lot safer if this is entirely handled inside the class, and `err_exit` can be passed as an argument to `ExpressionInputsValidator`.